packages/grid/_modules_/grid/models/rows.ts

```
@@ -44,9 +44,9 @@ export interface RowModel {
  * @param rowData Row as [[RowData]].
  * @returns A row as [[RowModel]].
  */
-export function createRow(rowData: RowData): RowModel {
+export function createRowModel(rowData: RowData, index: RowId): RowModel {
   const row: RowModel = {
-    id: rowData.id,
+    id: rowData.id == null ? index : rowData.id,
```

I think that we should throw if the id is missing to match the types.